pallets/api/src/messaging/benchmarking.rs

```diff
@@ -0,0 +1,274 @@
+				id: message_id,
+				query_id: 0,
+				response,
+			}
+			.into(),
+		);
+		assert!(Messages::<T>::get(&owner, &message_id).is_none());
+		assert!(XcmQueries::<T>::get(0).is_none());
+	}
+
+	/// x: Is it a get. (example: 1 = get, 0 = post)
+	/// y: the response has a callback.
+	#[benchmark]
+	fn ismp_on_response(x: Linear<0, 1>, y: Linear<0, 1>) {
+		let commitment = H256::repeat_byte(2u8);
+		let origin: T::AccountId = account("alice", 0, SEED);
+		let message_id = [1; 32];
+		let callback = if y == 1 {
+			// The mock will always assume successfull callback.
+			Some(Callback {
+				selector: [0; 4],
+				weight: 100.into(),
+				spare_weight_creditor: origin.clone(),
+				abi: Abi::Scale,
+			})
+		} else {
+			None
+		};
+
+		let message = Message::Ismp { commitment, callback, deposit: One::one() };
+
+		IsmpRequests::<T>::insert(&commitment, (&origin, &message_id));
+		Messages::<T>::insert(&origin, &message_id, &message);
+
+		let (response, event) = if x == 1 {
+			// get response
+			let get = ismp_get_response();
+			(
+				IsmpResponse::Get(get.clone()),
+				crate::messaging::Event::<T>::IsmpGetResponseReceived {
+					dest: origin,
+					id: message_id,
+					commitment,
+				},
+			)
+		} else {
+			// post response
+			let post = ismp_post_response();
+			(
+				IsmpResponse::Post(post.clone()),
+				crate::messaging::Event::<T>::IsmpPostResponseReceived {
+					dest: origin,
+					id: message_id,
+					commitment,
+				},
+			)
+		};
+
+		let handler = crate::messaging::ismp::Handler::<T>::new();
+
+		#[block]
+		{
+			handler.on_response(response.clone()).unwrap();
+		}
+
+		assert_has_event::<T>(event.into())
+	}
+
+	// Assuming a Timeout::Request(Request::Get) is handled the same as a
+	// Timeout::Request(Request::Post) x: Is a response timeout (example: 1 = response timeout, 0 =
+	// request timeout)
+	#[benchmark]
+	fn ismp_on_timeout(x: Linear<0, 1>) {
+		let commitment = H256::repeat_byte(2u8);
+		let origin: T::AccountId = account("alice", 0, SEED);
+		let message_id = [1; 32];
+		let message = Message::Ismp { commitment, callback: None, deposit: One::one() };
+
+		let timeout_message = if x == 1 {
+			Timeout::Response(ismp_post_response())
+		} else {
+			Timeout::Request(Request::Get(ismp_get_request()))
+		};
+
+		IsmpRequests::<T>::insert(&commitment, (&origin, &message_id));
+		Messages::<T>::insert(&origin, &message_id, &message);
+
+		let handler = crate::messaging::ismp::Handler::<T>::new();
+		#[block]
+		{
+			handler.on_timeout(timeout_message).unwrap();
+		}
+	}
+
+	#[benchmark]
+	fn ismp_get() {
+		let commitment = H256::repeat_byte(2u8);
+		let origin: T::AccountId = account("alice", 0, SEED);
+		let message_id = [1; 32];
+		let get = crate::messaging::ismp::Get::<T> {
+			dest: 0,
+			height: 0,
+			timeout: 0,
+			context: bounded_vec!(),
+			keys: bounded_vec!(),
+		};
+
+		#[extrinsic_call]
+		Pallet::<T>::ismp_get(RawOrigin::Signed(origin.clone()), message_id, get, One::one(), None);
```

Comment:
seems like a callback should be an option here as well.

Reply:
I haven't built this, but I believe compilation is going to fail. This PR removes the fee parameter, but you have `One::one()` in that place still.